skyllh/core/signal_generator.py

```diff
@@ -529,6 +583,113 @@ def _construct_signal_candidates(self):
             items=self._sig_candidates,
             probabilities=self._sig_candidates['weight'])
 
+    def _get_invalid_events_mask(
+            self,
+            events,
+            valid_event_field_ranges_dict,
+    ):
+        """Determines a boolean mask to select invalid events, which do not
+        fulfill the given valid event field ranges.
+
+        Parameters
+        ----------
+        events : instance of DataFieldRecordArray
+            The instance of DataFieldRecordArray of length N_events holding the
+            events to check.
+        valid_event_field_ranges_dict : dict
+            The dictionary holding the data field names (key) and their valid
+            value ranges (value).
+
+        Returns
+        -------
+        mask : instance of numpy.ndarray
+            The (N_events,)-shaped numpy.ndarray of bool, holding the mask of
+            the invalid events.
+        """
+        mask = np.zeros((len(events),), dtype=np.bool_)
+
+        for (field_name, min_max) in valid_event_field_ranges_dict.items():
+            if field_name not in events:
+                continue
```

Comment:
would be very nice to add a message to e.g. log.info of this decision, if possible

Reply:
We could also remove this check and then it would fail with a KeyError. Or actually require the field to be present and raise a descriptive error message if the field does not exist.

Reply:
I would prefer requiring the field to be present, as this would minimize the possibility in making a mistake when creating the `valid_event_field_ranges_dict` definition

Reply:
I added a KeyError if the field is missing.